### ISSUE TITLE
proc macro not expanded: failed to write request: Broken pipe (os error 32)

### ISSUE THREAD

**Issue by rohel01** [rust-lang/rust-analyzer#14289]
Hello,

This is similar to #12673, so I will provide similar information.

The diagnostic in VSCode

![image](https://user-images.githubusercontent.com/9367595/223782709-072b9acb-ad48-4e8a-ae5e-83a79db9d21f.png)

This is the Component proc-macro from bevy 0.10 BTW

In the VSCode "Rust Analyzer Language Server" output tab, I get the following message:
```
[ERROR rust_analyzer] Unexpected error: expected initialize request, got error: receiving on an empty and disconnected channel
expected initialize request, got error: receiving on an empty and disconnected channel
```

Infos from the terminal

```
$ rustup toolchain list
stable-x86_64-unknown-linux-gnu (default)
nightly-2023-03-07-x86_64-unknown-linux-gnu (override)
nightly-x86_64-unknown-linux-gnu

$ rustup --version
rustup 1.25.2 (17db695f1 2023-02-01)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.70.0-nightly (f63ccaf25 2023-03-06)`

$ rustc --version
rustc 1.70.0-nightly (f63ccaf25 2023-03-06)

$ /home/melchiore/.rustup/toolchains/nightly-2023-03-07-x86_64-unknown-linux-gnu/bin/rust-analyzer --version
rust-analyzer 1.70.0-nightly (f63ccaf 2023-03-06)
```

I am using vscode with an overriden rust-analyzer (as I tried to fix this issue).
Here is my `.vscode/settings.json`

``` yaml
{
    "rust-analyzer.procMacro.server": "/home/melchiore/.rustup/toolchains/nightly-2023-03-07-x86_64-unknown-linux-gnu/bin/rust-analyzer"
}
```

**Comment by lnicola** [comment]
> `    "rust-analyzer.procMacro.server": "/home/melchiore/.rustup/toolchains/nightly-2023-03-07-x86_64-unknown-linux-gnu/bin/rust-analyzer"`

That's not the right path, but rather `/home/melchiore/.rustup/toolchains/nightly-2023-03-07-x86_64-unknown-linux-gnu/libexec/rust-analyzer-proc-macro-srv`. But I don't recommend setting it, because we pick it up by default.

The LSP server path is `rust-analyzer.server.path`, but then again, you shouldn't change it.

**Comment by rohel01** [comment]
Hello,

Thanks for your answer. I tried to change these settings since I was getting a "Wrong ABI" error in VScode on proc macro expansion. Resetting everything to default value has fixed everything.
